### PR TITLE
chore: security toolchain bump, embedding diagnostics, dead-config cleanup

### PR DESCRIPTION
**Pull request by wcatz**
## Summary
Hardening batch from the full-codebase review:

- **Go 1.25.8 → 1.26.4** — govulncheck flagged 19 stdlib vulnerabilities under 1.25.8 (incl. net/http HTTP/2 CONTINUATION flood GO-2026-4918, crypto/tls key-update deadlock GO-2026-4870, several crypto/x509 DoS). Verified 0 findings under 1.26.4. CI builds from `go-version-file: go.mod`, so this propagates everywhere.
- **Embedding & linking diagnostics** — a missing Ollama model previously failed silently at debug level, leaving a production DB with zero embeddings (vector search + memory linking inert). Now:
  - `ghost mcp status` checks Ollama reachability, model presence (`/api/tags`), embedding coverage, and link counts
  - `ghost_health` reports embedded/total memories, warns on zero coverage or missing model with the exact `ollama pull` command, and shows link/scan counts
  - the embedding worker logs a WARN once per sweep on failures instead of per-item DEBUG
- **Dead config removal** — `api.model_quality` defaulted to a **nonexistent model ID** (`claude-opus-4-6-20250514`) and, along with `model_fast` and the entire `Defaults`/`Display`/`Server` sections, was referenced nowhere in the binary. Removed from struct, defaults, and tests. Extra YAML keys in existing user configs are silently ignored by koanf — not a breaking change.
- **UTF-8 truncation fix** — `ghost_memory_save`/`ghost_save_global` truncated content at a raw byte offset, which could split a multi-byte rune and store invalid UTF-8; now truncates at rune boundaries.
- **Docs** — README and docs/architecture.md now document the v0.9.0 memory-link layer; corrected the stale go:embed claim.

## Test plan
- [x] `go vet ./...`, full `go test ./...` (9 packages) under Go 1.26.4
- [x] govulncheck: 19 findings → 0
- [x] New tests: EmbeddingStats/LinkStats, HasModel (incl. tag-normalization cases), truncateUTF8 rune-boundary cases

## Summary by CodeRabbit

* **New Features**
  * Added memory linking/graph support, including related-memory connections and graph-aware search boosts.
  * Enhanced health/status checks to report embedding coverage, model availability, and link statistics.
  * Improved memory content truncation so multi-byte characters stay valid when content is shortened.

* **Bug Fixes**
  * Added clearer warnings when embedding models are missing or embedding/linking processing fails.
  * Improved configuration handling around embedding, reflection, and linking settings.

### CODE REVIEW

File: internal/mcpserver/mcpserver.go [hunk 1]
```diff
@@ -748,11 +757,30 @@ func (s *Server) registerTools() {
...
+				}
+			}
 		} else {
 			sb.WriteString("**Embeddings:** disabled\n")
 		}
 
+		if links, scans, err := s.store.LinkStats(ctx); err == nil {
+			fmt.Fprintf(&sb, "**Memory links:** %d links, %d memories scanned\n", links, scans)
+		}
+
```
[the remark this answers is not in the document]
Comment (wcatz) [reply internal/mcpserver/mcpserver.go:785]: Fixed in 31bd66f — HasModel errors are now reported instead of silently skipped.

File: README.md [hunk 2]
```diff
@@ -112,12 +113,14 @@ Add Ghost to your MCP config:
...
-Save → Search → Consolidate → Decay
+Save → Embed → Link → Search → Consolidate → Decay
 ```
 
 **Save** — Claude (or you) saves memories via MCP tools. Each memory has a category, importance score (0.0-1.0), and tags. FTS-based upsert deduplicates on save — if a similar memory already exists in the same category, it strengthens instead of creating a duplicate.
 
-**Search** — FTS5 full-text search with optional vector similarity (Ollama embeddings). Cross-project search finds knowledge from other repos. Global memories (`_global`) are included in every project's context.
+**Link** — A background worker connects related memories by embedding similarity (cosine ≥ 0.70) into a memory graph. Links self-heal after consolidation rewrites memories.
+
+**Search** — FTS5 full-text search with optional vector similarity (Ollama embeddings), plus a graph-expansion signal: memories linked to your top hits get boosted, surfacing related knowledge a keyword match would miss. Cross-project search finds knowledge from other repos. Global memories (`_global`) are included in every project's context.
```
[the remark this answers is not in the document]
Comment (wcatz) [reply README.md:125]: Fixed in 31bd66f — pipeline text now says six stages and the missing Embed paragraph was added.
